### PR TITLE
feat(SelectionInfo): Quick toggle or rotate for auras

### PR DESCRIPTION
**Pull request by SuikaXhq**
Replace the original quick edition of aura values with quick toggling and rotating.
RotationSlider is only shown when angle<360.

The toggling function is copied from TrackerSetting.vue. I thought it would be better if there is a unified function to access these states (also useful for mod API). It works at present though.

btw, I noticed that the orbit of RotationSlider's "slider" is actually larger than the rendered one. would that be a bug?

### CODE REVIEW

File: client/src/game/ui/SelectionInfo.vue [hunk 1]
```diff
@@ -69,6 +70,15 @@ function setValue(data: { solution: number; relativeMode: boolean }): void {
     }
 }
 
+function updateAura(aura: DeepReadonly<UiAura>, delta: Partial<Aura>, syncTo = true): void {
+    if (delta.value !== undefined && (isNaN(delta.value) || delta.value < 0)) delta.value = 0;
+    if (delta.dim !== undefined && (isNaN(delta.dim) || delta.dim < 0)) delta.dim = 0;
+    if (aura.temporary) {
+        auraSystem.add(aura.shape, { ...aura }, SERVER_SYNC);
+    }
```
Comment (Kruptein) [inline client/src/game/ui/SelectionInfo.vue:78]: I actually don't think you need to worry about these things. With the change you made, we're no longer modifying the `.value` or `.dim` and the temporary auras don't show up in the selection info either.

**Comment by Kruptein** [comment]
> The toggling function is copied from TrackerSetting.vue. I thought it would be better if there is a unified function to access these states (also useful for mod API). It works at present though.

I made a small comment on that, I understand you just copied it, but for the use here you can just call the auraSystem directly. The original code is specifically to handle a bunch of conditions when creating new auras.

> btw, I noticed that the orbit of RotationSlider's "slider" is actually larger than the rendered one. would that be a bug?

Yeah this is a bug, I'll fix this, I also noticed that the two sliders are not in sync (the quick toggle slider and the one in the full tracker settings), but that's also on me, I've fixed that locally already, but I'm gonna wait for this one to land first.

**Review by Kruptein — CHANGES_REQUESTED** [review]

File: client/src/game/ui/SelectionInfo.vue (hunk 1 above)
Comment (SuikaXhq) [reply client/src/game/ui/SelectionInfo.vue:78]: all right, I will remove that.
btw, what does "NO_SYNC" do actually? It seems to keep sync anyway.

Comment (Kruptein) [reply client/src/game/ui/SelectionInfo.vue]: Yeah so while the mouse is still pressed, the changes are only applied locally. When the mouse is released, it's properly synced to the server and all other players.

**Pull request merged**